File: src/main/java/io/github/binout/soccer/infrastructure/persistence/mongo/MongoConfiguration.java
package io.github.binout.soccer.infrastructure.persistence.mongo;

import com.mongodb.MongoClient;
import com.mongodb.MongoClientURI;
import com.mongodb.ServerAddress;
import com.mongodb.client.MongoDatabase;
import de.bwaldvogel.mongo.MongoServer;
import de.bwaldvogel.mongo.backend.memory.MemoryBackend;
import org.mongolink.MongoSessionManager;
import org.mongolink.Settings;
import org.mongolink.domain.mapper.ContextBuilder;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.util.StringUtils;

import java.net.InetSocketAddress;

@Configuration
public class MongoConfiguration {

    @Value("${app.mongodb.uri}")
    private String uri;

    @Bean
    public MongoDatabase database() {
        if (StringUtils.isEmpty(uri)) {
            MongoServer mongoServer = new MongoServer(new MemoryBackend());
            InetSocketAddress serverAddress = mongoServer.bind();
            MongoClient client = new MongoClient(new ServerAddress(serverAddress));
            return client.getDatabase("dev");
        } else {
            MongoClientURI mongoClientURI = new MongoClientURI(uri);
            MongoClient client = new MongoClient(mongoClientURI);
            return client.getDatabase(mongoClientURI.getDatabase());
        }
    }

    @Bean
    public MongoSessionManager sessionManager(MongoDatabase database) {
        ContextBuilder contextBuilder = new ContextBuilder(getClass().getPackage().getName());
        return MongoSessionManager.create(contextBuilder, Settings.defaultInstance().withDatabase(database));
    }

}

File: src/main/kotlin/io/github/binout/soccer/infrastructure/persistence/mongo/MongoConfiguration.kt
/*
 * Copyright 2016 Benoît Prioux
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.github.binout.soccer.infrastructure.persistence.mongo

import com.mongodb.MongoClient
import com.mongodb.MongoClientURI
import com.mongodb.ServerAddress
import com.mongodb.client.MongoDatabase
import de.bwaldvogel.mongo.MongoServer
import de.bwaldvogel.mongo.backend.memory.MemoryBackend
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.util.StringUtils

@Configuration
class MongoConfiguration(@Value("\${app.mongodb.uri}") private val uri: String) {

    @Bean
    fun database(): MongoDatabase = if (StringUtils.isEmpty(uri)) {
        val mongoServer = MongoServer(MemoryBackend())
        val serverAddress = mongoServer.bind()
        val client = MongoClient(ServerAddress(serverAddress))
        client.getDatabase("dev")
    } else {
        val mongoClientURI = MongoClientURI(uri!!)
        val client = MongoClient(mongoClientURI)
        client.getDatabase(mongoClientURI.database!!)
    }

}
